### PR TITLE
Matras integration

### PR DESCRIPTION
**Pull request by patkkka**

### CODE REVIEW

**Review by piotr921 — APPROVED** [review]

**Review by Dzionek95 — APPROVED** [review]

File: persistencebot/src/main/java/com/blackbook/persistencebot/service/BookDiscountParserService.java [hunk 1]
```diff
@@ -28,35 +28,37 @@
...
         if (book == null) {
             book = parsedBook;
         }
-        BookDiscount.setBook(book);
+        bookDiscount.setBook(book);
 
         //TODO remove if clause since creation data without bookstore id will not be permitted
         if (bookDiscountData.getBookstoreId() != null) {
-            BookDiscount.setBookstore(bookstoresRepository.findOne((long) bookDiscountData.getBookstoreId()));
+            bookDiscount.setBookstore(bookstoresRepository.findOne((long) bookDiscountData.getBookstoreId()));
```
Comment (Devas) [inline persistencebot/src/main/java/com/blackbook/persistencebot/service/BookDiscountParserService.java:43]: Maybe casting can be avoided?

File: pom.xml [hunk 2]
```diff
@@ -27,6 +27,7 @@
         <module>persistencebot</module>
         <module>models</module>
         <module>google_crawler</module>
+        <module>matras_scraper</module>
```
Comment (Devas) [inline pom.xml:30]: 👍

File: matras_scraper/src/main/java/com/blackbook/matrasscraper/scraper/Scraper.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                        .title(title)
+                        .subtitle(subtitle)
+                        .authors(authors)
+                        .genre(genre)
+                        .bookPageUrl(bookUrl)
+                        .coverUrl(coverUrl)
+                        .build())
+                .build();
+
+        String separator = " - ";
```
Comment (Devas) [inline matras_scraper/src/main/java/com/blackbook/matrasscraper/scraper/Scraper.java:94]: 👍

File: matras_scraper/src/main/java/com/blackbook/matrasscraper/scraper/Scraper.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        Scraper scraper = new Scraper(new JsoupHTMLDocumentProvider());
+        scraper.extractBookElements();
+    }
+
+    public List<BookDiscountData> extractBookElements() {
+        Document mainPageDoc = htmlDocumentProvider.provide(MATRAS_URL);
+        int lastPageNo = extractLastPageNo(mainPageDoc);
+        List<BookDiscountData> bookDiscountData = new LinkedList<>();
+        //TODO change 1 to property or lastPageNo if all pages wanted
+        for (int i = 0; i < 1; i++) {
```
Comment (Devas) [inline matras_scraper/src/main/java/com/blackbook/matrasscraper/scraper/Scraper.java:46]: ♻️

File: matras_scraper/src/main/java/com/blackbook/matrasscraper/controller/ScraperController.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+@RequestMapping(value = "/api/matras-scraper")
+public class ScraperController {
+
+    @Value("${endpoints.persistence-api}")
+    private String persistenceApiEndpoint;
+
+    @RequestMapping(method = RequestMethod.POST)
+    public List<BookDiscountView> postBookDiscounts() {
+        log.info("Transaction: POST /api/matras-scraper");
+        ClientHttpRequestFactory requestFactory = new BufferingClientHttpRequestFactory(new SimpleClientHttpRequestFactory()); //getClientHttpRequestFactory();
```
Comment (Devas) [inline matras_scraper/src/main/java/com/blackbook/matrasscraper/controller/ScraperController.java:35]: Remove comment

**Review by Devas — APPROVED** [review]
🐺

**Pull request merged**